### PR TITLE
Register Layout updates

### LINKED ISSUE #420: RegisterLayout updates
- Having the trap coordinate input before ordering stored under `RegisterLayout.trap_coordinates` is misleading because they can be confused with the ordered coordinates. They should be hidden under `RegisterLayout._trap_coordinates`
- There should be a tutorial on how to work with `RegisterLayout` added to the docs

### PR DESCRIPTION
**Pull request by HGSilveri**
Closes #420 #414 .

### CODE REVIEW

**Comment by HGSilveri** [comment]
I just noticed that the code highlighting in the notebooks is broken. This seems to be related to the latest update of `ipython`, as mentioned [here](https://github.com/ipython/ipython/issues/13845). 

For now, I'll just avoid that version.

File: tutorials/advanced_features/Register Layouts.ipynb [hunk 1]
```diff
@@ -0,0 +1,551 @@
...
+    "To access the trap coordinates:\n",
+    "- `RegisterLayout.traps_dict` gives a mapping between trap IDs and coordinates\n",
+    "- `RegisterLayout.coords` provides the ordered list of trap coordinates"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To identify a layout, one can use it's `repr()` for a unique identifier or its `str()` for the `slug` (if specified)."
```
Comment (CdeTerra) [inline tutorials/advanced_features/Register Layouts.ipynb:100]: Typo:
```suggestion
    "To identify a layout, one can use its `repr()` for a unique identifier or its `str()` for the `slug` (if specified)."
```

**Review by CdeTerra — APPROVED** [review]
Great tutorial!

File: tutorials/advanced_features/Register Layouts.ipynb (hunk 1 above)
Comment (HGSilveri) [reply tutorials/advanced_features/Register Layouts.ipynb:100]: Thanks! Good catch, these typos are usually my nemesis...

**Pull request merged**